### PR TITLE
Feature/vr 247

### PR DESCRIPTION
**Pull request by n3op2**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [x] Feature

## Linked tickets

https://digicatapult.atlassian.net/jira/software/c/projects/ENG/boards/206?selectedIssue=VR-247&sprints=959

## High level description

adding another persona and dependencies such as database, agent, ui.

Screenshot 2024-10-04 at 13 19 36
Screenshot 2024-10-04 at 13 18 12
Screenshot 2024-10-04 at 13 16 55

## Detailed description

Some logs from a manual test
```sh
postgres-veritable-cloudagent-charlie  | 2024-10-04 11:06:50.324 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres-veritable-cloudagent-charlie  | 2024-10-04 11:06:50.327 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres-veritable-cloudagent-charlie  | 2024-10-04 11:06:50.332 UTC [29] LOG:  database system was shut down at 2024-10-04 10:39:12 UTC
postgres-veritable-cloudagent-charlie  | 2024-10-04 11:06:50.337 UTC [1] LOG:  database system is ready to accept connections
postgres-veritable-cloudagent-charlie  | 2024-10-04 11:11:50.332 UTC [27] LOG:  checkpoint starting: time
postgres-veritable-cloudagent-charlie  | 2024-10-04 11:11:50.338 UTC [27] LOG:  checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.002 s, sync=0.001 s, total=0.007 s; sync files=2, longest=0.001 s, average=0.001 s; distance=0 kB, estimate=0 kB; lsn=0/1D9E950, redo lsn=0/1D9E8F8
...
veritable-ui-charlie                   | added 17 packages in 3s
veritable-ui-charlie                   | 
veritable-ui-charlie                   | 3 packages are looking for funding
veritable-ui-charlie                   |   run `npm fund` for details
veritable-ui-charlie                   | Using environment: production
veritable-ui-charlie                   | Already up to date
veritable-ui-charlie                   | 
veritable-ui-charlie                   | > veritable-ui@0.9.5 start
veritable-ui-charlie                   | > node build/index.js
veritable-ui-charlie                   | 
veritable-ui-charlie                   | 12:16:43 ✨ veritable-ui For issuing credentials using:
veritable-ui-charlie                   | 	issuerId:	did:key:z6Mkg73TEYVn3Ukh9swqK6nReJjLP85yEgyv6xEJR6aWuaGb
veritable-ui-charlie                   | 	schemaId:	ipfs://bafkreihjmsf7x4bp5jzsvm6zfyjvrig6htxnuz3742rtnmc45kxxwi47bq
veritable-ui-charlie                   | 	cred-def:	ipfs://bafkreidzq3ti3z5opwfmkm3dq6genc7jfyx4nasdnjalhymqn66ypozcem
veritable-ui-charlie                   | 12:16:43 ✨ veritable-ui WebSocket connection to Cloudagent established
veritable-ui-charlie                   | 12:16:43 ✨ veritable-ui htmx-tsoa listening on 3000 port
veritable-ui-charlie                   | 12:16:51 🐛 veritable-ui returning connections page {"connections":[],"search":""}
veritable-ui-charlie                   | 12:16:51 ✨ veritable-ui request completed GET 200 /connection?search= 117ms
veritable-ui-charlie                   | 12:16:52 🐛 veritable-ui returning connections page {"connections":[],"search":""}
veritable-ui-charlie                   | 12:16:52 ✨ veritable-ui request completed GET 200 /connection 17ms
veritable-ui-charlie                   | 12:16:52 ✨ veritable-ui request completed GET 304 /lib/htmx.org/htmx.min.js 4ms
veritable-ui-charlie                   | 12:16:52 ✨ veritable-ui request completed GET 304 /public/styles/mpa.css 3ms
veritable-ui-charlie                   | 12:16:52 ✨ veritable-ui request completed GET 304 /public/images/logo-square.svg 2ms
veritable-ui-charlie                   | 12:16:52 ✨ veritable-ui request completed GET 304 /lib/htmx-ext-json-enc/json-enc.js 3ms
veritable-ui-charlie                   | 12:16:52 ✨ veritable-ui request completed GET 304 /public/styles/shared.css 5ms
smtp4dev                               | Overriding HTTP_PORTS '80' and HTTPS_PORTS ''. Binding to values defined by URLS instead 'http://*:80'.
veritable-ui-charlie                   | 12:16:52 ✨ veritable-ui request completed GET 304 /public/styles/main.css 4ms
veritable-ui-charlie                   | 12:16:52 ✨ veritable-ui request completed GET 304 /public/scripts/auth-redirect.js 1ms
...
veritable-cloudagent-charlie           | {"level":40,"time":1728040017333,"pid":1,"hostname":"f203cef032b5","name":"veritable-cloudagent","component":"credo-ts-agent","msg":"The 'SdJwtVc' module is experimental and could have unexpected breaking changes. When using this module, make sure to use strict versions for all @credo-ts packages."}
veritable-cloudagent-charlie           | {"level":40,"time":1728040017334,"pid":1,"hostname":"f203cef032b5","name":"veritable-cloudagent","component":"credo-ts-agent","msg":"The 'X509' module is experimental and could have unexpected breaking changes. When using this module, make sure to use strict versions for all @credo-ts packages."}
veritable-ui-charlie                   | 12:17:12 🐛 veritable-ui storing VERITABLE_NONCE cookie 77rqBkf15rP1gWqpJKt-uA
veritable-ui-charlie                   | 12:17:12 🐛 veritable-ui storing VERITABLE_REDIRECT cookie 77rqBkf15rP1gWqpJKt-uA
veritable-cloudagent-charlie           | {"level":30,"time":1728040017334,"pid":1,"hostname":"f203cef032b5","name":"veritable-cloudagent","component":"credo-ts-agent","msg":"Creating agent with config {\"agentConfig\":{\"label\":\"vertiable-cloudagent\",\"logger\":3,\"walletConfig\":{\"id\":\"charlie\",\"key\":\"[*****]\",\"storage\":{\"type\":\"postgres\",\"config\":{\"host\":\"postgres-veritable-cloudagent-charlie:5432\"},\"credentials\":\"[*****]\"}},\"endpoints\":[\"ws://veritable-cloudagent-charlie:5003\"],\"connectionImageUrl\":\"https://image.com/image.png\",\"backupBeforeStorageUpdate\":false,\"autoUpdateStorageOnStartup\":true,\"useDidKeyInProtocols\":true,\"useDidSovPrefixWhereAllowed\":false,\"agentDependencies\":true}}"}
veritable-cloudagent-charlie           | {"level":30,"time":1728040017347,"pid":1,"hostname":"f203cef032b5","name":"veritable-cloudagent","component":"credo-ts-agent","msg":"Initializing wallet 'charlie' {\"id\":\"charlie\",\"key\":\"[*****]\",\"storage\":{\"type\":\"postgres\",\"config\":{\"host\":\"postgres-veritable-cloudagent-charlie:5432\"},\"credentials\":\"[*****]\"}}"}
veritable-cloudagent-charlie           | {"level":30,"time":1728040018375,"pid":1,"hostname":"f203cef032b5","name":"veritable-cloudagent","component":"credo-ts-agent","msg":"Agent storage is up to date."}
veritable-cloudagent-charlie           | {"level":30,"time":1728040018414,"pid":1,"hostname":"f203cef032b5","name":"veritable-cloudagent","msg":"Successfully started server on port 3000"}
veritable-ui-charlie                   | 12:17:12 🐛 veritable-ui login redirect to http://localhost:3080/realms/veritable/protocol/openid-connect/auth?response_type=code&client_id=veritable-ui&redirect_uri=http%3A%2F%2Flocalhost%3A3002%2Fauth%2Fcallback&state=77rqBkf15rP1gWqpJKt-uA.KxSKW1v2SJa68Vp_-M0k7R_48sSryGlDSUyxaB8p6fk&scope=openid
veritable-ui-charlie                   | 12:17:12 ✨ veritable-ui request completed GET 302 /auth/login?path=%2Fqueries 6ms
veritable-ui-charlie                   | 12:17:12 🐛 veritable-ui Login callback will redirect to http://localhost:3002/auth/redirect?state=77rqBkf15rP1gWqpJKt-uA.KxSKW1v2SJa68Vp_-M0k7R_48sSryGlDSUyxaB8p6fk&code=8b9f0469-2a1b-4e47-b949-2fb12af36ef4.d072447a-8c7c-4aeb-a3c6-d28b5c882315.d48dc130-3ef6-458e-9cd3-553f395950a1
veritable-ui-charlie                   | 12:17:12 ✨ veritable-ui request completed GET 200 /auth/callback?state=77rqBkf15rP1gWqpJKt-uA.KxSKW1v2SJa68Vp_-M0k7R_48sSryGlDSUyxaB8p6fk&session_state=d072447a-8c7c-4aeb-a3c6-d28b5c882315&iss=http%3A%2F%2Flocalhost%3A3080%2Frealms%2Fveritable&code=8b9f0469-2a1b-4e47-b949-2fb12af36ef4.d072447a-8c7c-4aeb-a3c6-d28b5c882315.d48dc130-3ef6-458e-9cd3-553f395950a1 1ms
veritable-ui-charlie                   | 12:17:12 🐛 veritable-ui resetting cookies: VERITABLE_NONCE, VERITABLE_REDIRECT, VERITABLE_ACCESS_TOKEN, VERITABLE_REFRESH_TOKEN
```

to be updated...

## Operational impact

none - since it's only local dev focused

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -140,6 +140,92 @@ services:
...
+    restart: always
+    depends_on:
+      ipfs:
+        condition: service_healthy
+    ports:
+      - 3102:3000
+    env_file:
+      - docker/cloudagent.env
+    environment:
+      - ENDPOINT=ws://veritable-cloudagent-charlie:5003
```
Comment (rmlearney-digicatapult) [inline docker-compose.yml:224]: Do all the agents' websocket ports point to 5003?

Comment (n3op2) [reply docker-compose.yml:224]: yes! also a new observation.

Comment (rmlearney-digicatapult) [reply docker-compose.yml:224]: smtp4dev is on 5001 and 5000 is used by Apple AirPlay if you're going to change them

Comment (mattdean-digicatapult) [reply docker-compose.yml:224]: yep all 5003 intentionally

**Review by mattdean-digicatapult — DISMISSED** [review]
lgtm as long as you've done the manual connection test

**Review by mattdean-digicatapult — APPROVED** [review]
lgtm

**Pull request merged**